executor/executor.go

```diff
@@ -1488,6 +1466,62 @@ func (e *SortExec) Next() (*Row, error) {
 	return row, nil
 }
 
+type TopnExec struct {
+	SortExec
+	limit      *plan.Limit
+	resultRows []*orderByRow
+}
+
+func (e *TopnExec) Next() (*Row, error) {
+	if !e.fetched {
+		e.Idx = 1 + int(e.limit.Offset)
+		totalCount := int(e.limit.Offset + e.limit.Count)
+		e.Rows = make([]*orderByRow, 1, totalCount+2)
```

Comment:
It's a trick: it's easy to calculate index by setting first index to 1, then i's left child must be 2 * i, right child must be 2 * i + 1, i's parent must be i / 2